File: src_local/agents/Python/bridge.py
from src_local.agents.base import BaseActivePersona
import logging
import time

logger = logging.getLogger(__name__)

class BridgePersona(BaseActivePersona):
    """Core: PhD in Distributed Systems 🌉"""
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Bridge", "🌉", "PhD Systems Architect", "Python"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Camadas de Interoperabilidade...")
        
        audit_rules = [
            {'regex': r'subprocess\.run\(.*shell=True', 'issue': 'Vulnerabilidade: Shell Injection.', 'severity': 'critical'},
            {'regex': r'os\.system\(', 'issue': 'Obsolescência: Use subprocess.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.py',), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        # O Bridge delega a detecção de risco para o AuditEngine via perform_audit
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, mestre em integração."

File: src_local/agents/Bun/System/bridge.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * 🌉 Dr. Bridge — PhD in Bun FFI & Native Integration
 * Especialista em Bun FFI, integração nativa e contratos de API.
 */
export class BridgePersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Bridge";
        this.emoji = "🌉";
        this.role = "PhD Bun Integration Architect";
        this.phd_identity = "Native Integration & Bun FFI Strategy";
        this.stack = "Bun";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        return findings;
    }

    override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.ts', '.tsx'],
            rules: [
                { regex: /bun:ffi/, issue: 'FFI: Uso de bun:ffi para integração nativa — verifique memory safety PhD.', severity: 'medium' },
                { regex: /dlopen\(/, issue: 'FFI: dlopen carregando biblioteca nativa — verifique origem e permissões PhD.', severity: 'high' },
                { regex: /fetch\s*\([^)]*\)\.then\([^)]*as\s+any/, issue: 'Contrato Quebrado: Resposta fetch tipada como "any" no Bun PhD.', severity: 'high' },
                { regex: /Bun\.file\([^)]+\)\.json\(\)(?!\s*as\b)/, issue: 'Contrato Vago: Bun.file().json() sem tipagem PhD.', severity: 'medium' },
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && /bun:ffi|dlopen/.test(content)) {
            return {
                file, severity: "HIGH",
                issue: `Risco de Integração: O objetivo '${objective}' exige safety. Em '${file}', FFI nativo requer verificação de memory safety no Bun PhD.`,
                context: "FFI usage detected"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Bridge (Bun): Analisando blindagem de contratos FFI para ${objective}.`,
            context: this.name
        };
    }

    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em integração nativa e FFI Bun.`;
    }
}
